src/storage/accessors/FileDataAccessor.ts

```diff
@@ -159,8 +159,14 @@ export class FileDataAccessor implements DataAccessor {
    */
   private async getFileMetadata(link: ResourceLink, stats: Stats):
   Promise<RepresentationMetadata> {
-    return (await this.getBaseMetadata(link, stats, false))
-      .set(CONTENT_TYPE_TERM, link.contentType);
+    const metadata = await this.getBaseMetadata(link, stats, false);
+    // If the resource is using an unsupported contentType, the original contentType was written to the metadata file.
+    // As a result, we should only set the contentType derived from the file path,
+    // when no previous metadata entry for contentType is present.
+    if (metadata.contentType === undefined) {
```

Comment:
```suggestion
    if (typeof metadata.contentType === 'undefined') {
```
Very minor change. In theory there are potential situations where someone modified the `undefined` object.